runtime/compiler/ilgen/Walker.cpp

```diff
@@ -5061,13 +5061,57 @@ TR_J9ByteCodeIlGenerator::loadAuto(TR::DataType type, int32_t slot, bool isAdjun
    push(load);
    }
 
+static bool isFieldResovled(TR::Compilation *comp, TR_ResolvedJ9Method * owningMethod, int32_t cpIndex, bool isStore)
+   {
+   uint32_t offset = 0;
+   TR::DataType type = TR::NoType;
+   bool isVolatile = true, isFinal = false, isPrivate = false, isUnresolvedInCP;
+   bool bResolved = owningMethod->fieldAttributes(comp, cpIndex, &offset, &type, &isVolatile, &isFinal,
+                                    &isPrivate, isStore, &isUnresolvedInCP, true /* needsAOTValidation */);
+   return bResolved;
```

Comment:
I think this should actually be
```suggestion
   return !isUnresolvedInCP;
```

`fieldAttributes()` will return true if the CP entry gets resolved at compile-time. Under such cases, the JIT basically gets to peek at what the result will be when the entry is resolved. However, the runtime CP resolution still has to happen, so the JIT needs to treat the field as unresolved (because it is) even though it actually knows what it's supposed to be.

Reply:
Thanks for the clarification!

Reply:
Actually I have a question on this code. For loadInstance, should it check if a field is resolved or if its containing class is resolved. Here on checking if the field is resolved I used how `findOrCreateShadowSymbol()` does it as blow: 

https://github.com/eclipse/openj9/blob/141cd78d2ad744ca44f04b97d0c3927ee854eaf9/runtime/compiler/compile/J9SymbolReferenceTable.cpp#L800

Looking at `genWithField()`, it checks if the containing class exists or not to determine resolve or not.
https://github.com/eclipse/openj9/blob/141cd78d2ad744ca44f04b97d0c3927ee854eaf9/runtime/compiler/ilgen/Walker.cpp#L6153-L6157

Reply:
`isFieldResovled()` will be removed from this PR and added to runtime helper PR.